crates/cairo-lint-core/src/plugin.rs

```diff
@@ -82,6 +84,12 @@ impl AnalyzerPlugin for CairoLint {
                         &mut diags,
                     ),
                     SyntaxKind::StatementBreak => breaks::check_break(db.upcast(), node, &mut diags),
+                    SyntaxKind::ExprBinary => {
+                        // Convertir el nodo a `ExprBinary` utilizando `match`
+                        if let AstExpr::Binary(expr_binary) = AstExpr::from_syntax_node(db.upcast(), node) {
+                            equal_operand::check_redundant_comparison(db.upcast(), &expr_binary, &mut diags);
+                        }
+                    }
```

Comment:
I don't think you should check using the syntax nodes, add the check in the iteration above, similarly to [that](https://github.com/keep-starknet-strange/cairo-lint/blob/e6a73108dc281b78533088c9e16700e3d3ca8230/crates/cairo-lint-core/src/plugin.rs#L51-L56). Then you'll need to check the `cairo_lang_semantic::Expr::If`. check [this](https://github.com/keep-starknet-strange/cairo-lint/blob/main/crates/cairo-lint-core/src/lints/loops.rs) and [this](https://github.com/keep-starknet-strange/cairo-lint/blob/main/crates/cairo-lint-core/src/lints/single_match.rs) for inspiration. Let me know if that helps or not

Reply:
I see. It seems that I have to change almost everything if I'm not wrong.